mesa/space.py

```diff
@@ -246,11 +245,15 @@ def get_neighbors(self, pos, moore,
         return list(self.iter_neighbors(
             pos, moore, include_center, radius))
 
-    def torus_adj(self, coord, dim_len):
+    def torus_adj(self, pos):
         """ Convert coordinate, handling torus looping. """
-        if self.torus:
-            coord %= dim_len
-        return coord
+        if not self.out_of_bounds(pos):
+            return pos
+        elif not self.torus:
+            raise Exception("Point out of bounds, and space non-toroidal.")
```

Comment:
Raise a bare `Exception`, while valid, is not very descriptive, and requires writing a message within the specific implementation.

If would be great if we could add an `AgentOutOfBoundsError` exception. So adding something like this to `mesa/space.py`:
```
# New Exception class
class AgentOutOfBoundsError(Exception):
        def __init__(self, position):
                super().__init__(self, "Agent position {} is out of bounds.".format(position))

# Usage
def torus_adj(self, pos):
...
        elif not self.torus:
                raise AgentOutOfBoundsError(pos)

Reply:
I had considered this, but didn't do since it didn't directly address the bug.

Reply:
@JamesArruda that is fair enough. I think it's more important at this point to get the bug fix in. I can submit a PR that adds the `AgentOutOfBoundsError`.